ros2/ament.bzl

```diff
@@ -390,3 +391,89 @@ def sh_launcher(name, deps, idl_deps = None, **kwargs):
         ament_setup = ament_setup,
         **kwargs
     )
+
+def _ros2_ament_cpp_library_impl(ctx):
+    src_output = ctx.actions.declare_file(ctx.attr.basename + ".cpp")
+    hdr_output = ctx.actions.declare_file(ctx.attr.basename + ".hpp")
```

Comment:
The good practice is to create a folder with the label name and put the files therein. Here:
```
src_output = ctx.actions.declare_file(ctx.attr.name + "/ament_setup.cc")
hdr_output = ctx.actions.declare_file(ctx.attr.name + "/ament_setup.h")
```

Reply:
I'm using the name of the wrapper target (basename) to avoid the `_srcs` suffix of the implementation target leaking into the generated code.